frontend/src/pages/surveys/components/preferences-step/preferences-step.tsx

```diff
@@ -0,0 +1,108 @@
+import { type ControllerRenderProps } from 'react-hook-form';
+
+import { Button, Input, SurveyCategory } from '#libs/components/components.js';
+import {
+  useAppForm,
+  useCallback,
+  useFormController,
+} from '#libs/hooks/hooks.js';
+import {
+  type SurveyInputDto,
+  surveyInputValidationSchema,
+  SurveyTextareaOptions,
+} from '#packages/survey/survey.js';
+import {
+  DEFAULT_SURVEY_PAYLOAD,
+  PREFERENCES_CATEGORIES,
+  TEXTAREA_MAX_LENGTH,
+  TEXTAREA_ROWS_COUNT,
+} from '#pages/surveys/libs/constants.js';
+
+import styles from './styles.module.scss';
+
+type Properties = {
+  onSubmit: (options: string[]) => void;
+};
+
+const PreferencesStep: React.FC<Properties> = ({ onSubmit }) => {
+  const { watch, control, errors, isValid, handleSubmit } =
+    useAppForm<SurveyInputDto>({
+      defaultValues: DEFAULT_SURVEY_PAYLOAD,
+      validationSchema: surveyInputValidationSchema,
+    });
+  const { field: optionsField } = useFormController({
+    name: 'options',
+    control,
+  });
+
+  const options: string[] = [
+    ...PREFERENCES_CATEGORIES,
+    SurveyTextareaOptions.OTHER,
+  ];
+
+  const activeOptions = watch('options');
+  const hasOther = activeOptions.includes('Other');
+
+  const handleOnSubmit = useCallback(
```

Comment:
```suggestion
  const handleSubmit = useCallback(
```

Reply:
i guess its better not to change naming here, because we already have handleSubmit here:  const { watch, control, errors, isValid, handleSubmit } =
    useAppForm<SurveyInputDto>

Reply:
Missed handleSubmit from useAppForm. Let's not rename handleSubmit that comes from useAppForm and name the other function handlePreferencesSubmit

Reply:
Done